test/utils_test.js

```diff
@@ -511,20 +511,59 @@ describe("utils", () => {
       };
 
       expect(toIdSchema(schema)).eql({
-        id: "root",
+        "$id": "root",
         level1a: {
-          id: "root_level1a",
-          level1a2a: {id: "root_level1a_level1a2a"},
-          level1a2b: {id: "root_level1a_level1a2b"},
+          "$id": "root_level1a",
+          level1a2a: {"$id": "root_level1a_level1a2a"},
+          level1a2b: {"$id": "root_level1a_level1a2b"},
         },
         level1b: {
-          id: "root_level1b",
-          level1b2a: {id: "root_level1b_level1b2a"},
-          level1b2b: {id: "root_level1b_level1b2b"},
+          "$id": "root_level1b",
+          level1b2a: {"$id": "root_level1b_level1b2a"},
+          level1b2b: {"$id": "root_level1b_level1b2b"},
         },
       });
     });
 
+    it("schema with an id property must not corrupt the idSchema", () => {
+      const schema = {
+        title: "User",
+        description: "Application user",
+        type: "object",
+        properties: {
+          metadata: {
+            type: "object",
+            properties: {
+              id: {
+                type: "string"
+              },
+              name: {
+                type: "string"
+              },
+              sex: {
+                type: "string",
+                enum: [ "M", "F", "Other" ]
+              },
+              age: {
+                type: "integer"
+              }
+            },
+            required: [ "id", "age", "name" ]
+          }
+        }
```

Comment:
I don't think we need such a large sample schema; a simple schema like

```js
const schema = {
  type: "object",
  properties: {
    id: {type: "string"}
  }
};
```

is good enough.